docs/DeveloperGuide.md

```diff
@@ -1,34 +1,369 @@
+By: CS2113-T09-4    Since: September 2020   Licence: MIT
 
+<br>
 
-## Product scope
-### Target user profile
+## Table of Contents
+
+<br>
+
+* Table of Contents
+{:toc}
+
+## 1.0 Setting up
+
+### 1.1 Prerequisites
+ * Java 11 (can be download from [here](https://docs.aws.amazon.com/corretto/latest/corretto-11-ug/downloads-list.html))
+ * Intellij IDE
+ 
+<br>
+    
+### 1.2 Setting up:    
+* Download the latest **traKCAL** jar file ([here](https://github.com/AY2021S1-CS2113T-T09-4/tp/releases))
+* Copy jar file into an empty folder in a convenient location (eg. Desktop)
+* Open terminal and navigate directory to the folder containing jar file 
+* Enter the following command line to run program: ```java -jar trakCAL.jar```   
+* Look through the user guide for a full detailed explanation on the functionality of **trakCAL**
+    
+<br>
+<br>    
+    
+## 2.0 Design 
+
+### 2.1 Architecture
+
+![Architecture](diagrams/Architecture.png)
+
+*Figure 1. Diagram for Model component*
+
+The Architecture Diagram shown above explains the high-level design of **trakCAL**.
+
+**trakCAL** is made up of mainly 6 components.
+
+`trakCAL`: tracKCAL starts here
+
+`Ui`: Displays any message that the user can see and interact with
+
+`Logic`: Interprets what the user input
+
+`Command`: Executes specific command according to interpretation by `Logic`
+
+`Storage`: Saves required data into the hard disk or retrieves data
+
+`Model`: Visualize data into graph
+
+<br>
+
+### 2.2 Logic component
+
+![Logic Component](diagrams/LogicComponent.png)
+
+*Figure 2. Diagram for logic component*
+
+The main bulk of data processing takes place in the logic component. In this component, 
+the data from the userinput is checked for its validity and parsed down futher to its respective command blocks.
+These blocks of command are derived from the abstract Command class. Each different command block deals with the 
+proposed functionality which can be associated with `Ui`,`storage` or `model` components. 
+
+* **trakCAL** uses the parser class to filter based on command words by user.
+* Description that comes after is further parsed down by the Parser.
+* Respective Command object is created and is executed by **trakCAL**.
+* Respective execution methods can be further associated with `UI`, `Storage` and `Model` components.
+
+<br>
+
+### 2.3 Model component
+
+![Model_Component](diagrams/model.png)
+
+*Figure 3. Diagram for Model component*
+
+**API**: Model.java 
+
+In the Model component, 
+
+* Stores a DayMap object that holds the data for each date.
+* Stores an ActivityList that holds the list of Activities, Food or Exercise for each day 
+* Does not depend on any of the other components.
+
+<br>
+<br>
+
+## 3.0 Implementation
+
+### 3.1 Create User Profile
+
+The sequence diagram below shows how the components will react to a new user or for a returning user. 
+
+![CreateNewUser](diagrams/CreateNewUser.png)
```

Comment:
![image](https://user-images.githubusercontent.com/46754292/97525874-d1e9dd00-19e2-11eb-80d3-4fe11b5ab1ef.png)
Is there an error in the red circle? Should the incoming arrow from the left go to the name box as it is a constructor? And which function calls the data for user profile function?